commands/command_test.go

```diff
@@ -962,6 +963,94 @@ func (s *MmctlUnitTestSuite) TestCommandShowCmd() {
 		s.Require().Len(printer.GetErrorLines(), 0)
 	})
 
+	s.Run("Show custom slash command via team:trigger", func() {
+		printer.Clean()
+
+		list := []*model.Command{copyCommand(&mockCommand), &mockCommand, copyCommand(&mockCommand)}
+		list[0].Trigger = "bloop"
+		list[2].Trigger = "bleep"
+
+		s.client.
+			EXPECT().
+			GetTeamByName(mockTeam.Name, "").
+			Return(&mockTeam, &model.Response{Error: nil}).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			ListCommands(mockTeam.Id, false).
+			Return(list, &model.Response{Error: nil}).
+			Times(1)
+
+		err := showCommandCmdF(s.client, &cobra.Command{}, []string{fmt.Sprintf("%s:%s", mockTeam.Name, mockCommand.Trigger)})
+		s.Require().NoError(err)
+		s.Require().Len(printer.GetLines(), 1)
+		s.Equal(&mockCommand, printer.GetLines()[0])
+		s.Require().Len(printer.GetErrorLines(), 0)
+	})
+
+	s.Run("Show custom slash command via team:trigger with invalid team", func() {
+		printer.Clean()
+
+		list := []*model.Command{copyCommand(&mockCommand), &mockCommand, copyCommand(&mockCommand)}
+		list[0].Trigger = "bloop"
+		list[2].Trigger = "bleep"
+
+		const teamName = "bogus_team"
+		teamTrigger := fmt.Sprintf("%s:%s", teamName, mockCommand.Trigger)
+
+		s.client.
+			EXPECT().
+			GetTeamByName(teamName, "").
+			Return(nil, &model.Response{Error: &model.AppError{Message: "team not found"}}).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			GetCommandById(teamTrigger).
+			Return(nil, &model.Response{Error: &model.AppError{Message: "command not found"}}).
+			Times(1)
+
+		err := showCommandCmdF(s.client, &cobra.Command{}, []string{teamTrigger})
+		s.Require().Error(err)
+		s.Require().Len(printer.GetLines(), 0)
+		s.Require().Len(printer.GetErrorLines(), 0)
+	})
+
+	s.Run("Show custom slash command via team:trigger with invalid trigger", func() {
+		printer.Clean()
+
+		list := []*model.Command{copyCommand(&mockCommand), &mockCommand, copyCommand(&mockCommand)}
+		list[0].Trigger = "bloop"
+		list[2].Trigger = "bleep"
+
+		const trigger = "bogus_trigger"
+		teamTrigger := fmt.Sprintf("%s:%s", mockTeam.Name, trigger)
+
+		s.client.
+			EXPECT().
+			GetTeamByName(mockTeam.Name, "").
+			Return(&mockTeam, &model.Response{Error: nil}).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			ListCommands(mockTeam.Id, false).
+			Return(list, &model.Response{Error: nil}).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			GetCommandById(teamTrigger).
+			Return(nil, &model.Response{Error: &model.AppError{Message: "command not found"}}).
+			Times(1)
+
+		err := showCommandCmdF(s.client, &cobra.Command{}, []string{teamTrigger})
+		s.Require().Error(err)
+		s.Require().Len(printer.GetLines(), 0)
+		s.Require().Len(printer.GetErrorLines(), 0)
```

Comment:
Maybe we can check error string with `Equal` or `Contains` to see if have the right error.

Reply:
I added this https://github.com/mattermost/mmctl/pull/135/commits/958a7c19f8a20e5e4ebc4329e4e91283fe864f5b , but I must admit I'm torn regarding how this makes the units test very brittle versus the benefit.  Ongoing I think this concept should be discussed decided via the team.